src/arns.lua

```diff
@@ -423,14 +424,106 @@ function arns.assertValidIncreaseUndername(record, qty, currentTimestamp)
 	return true
 end
 
+-- AUCTIONS
+
+--- Creates an auction for a given name
+--- @param name string The name of the auction
+--- @param timestamp number The timestamp to start the auction
+--- @param initiator string The address of the initiator of the auction
+--- @return Auction|nil The auction instance
+function arns.createAuction(name, timestamp, initiator)
+	if not arns.getRecord(name) then
+		error("Name is not registered. Auctions must be created for registered names.")
+	end
+	if arns.getAuction(name) then
+		error("Auction already exists for name")
+	end
+
+	local baseFee = demand.getFees()[#name]
+	local demandFactor = demand.getDemandFactor()
+	local auction = Auction:new(name, timestamp, demandFactor, baseFee, initiator, arns.calculateRegistrationFee)
+	NameRegistry.auctions[name] = auction
+	-- ensure the name is removed from the registry
+	arns.removeRecord(name)
+	return auction
+end
+
+function arns.getAuction(name)
+	return NameRegistry.auctions[name]
+end
+
+function arns.getAuctions()
+	return NameRegistry.auctions or {}
+end
```

Comment:
the `or {}` implies auctions could be undefined which would indicate that `NameRegistry.auctions[name]` could attempt to index a nil value

Reply:
it can be, if we don't update the state manually

Reply:
>  if we don't update the state manually
Is that the plan?